pkg/operation/botanist/component/extensions/dnsrecord/dnsrecord_test.go

```diff
@@ -341,6 +343,38 @@ var _ = Describe("DNSRecord", func() {
 				Expect(deployedDNS).To(DeepEqual(expectedDNSRecord))
 			})
 
+			It("should set reconcile operation annotation on dnsrecord", func() {
+				delete(dns.Annotations, v1beta1constants.GardenerOperation)
+
+				// set AnnotateOperation and ReconcileOnlyOnChangeOrError to false
+				values.AnnotateOperation = false
+				values.ReconcileOnlyOnChangeOrError = false
+
+				resourceVersion, err := strconv.Atoi(expectedDNSRecord.ResourceVersion)
+				Expect(err).NotTo(HaveOccurred())
+
+				expectedDNSRecord.ResourceVersion = strconv.Itoa(resourceVersion + 1)
+				expectedDNSRecord.Annotations = map[string]string{
+					v1beta1constants.GardenerTimestamp: now.UTC().String(),
+					v1beta1constants.GardenerOperation: v1beta1constants.GardenerOperationReconcile,
+				}
+				expectedDNSRecord.Spec.Values = []string{address, "8.8.8.8", "1.1.1.1"}
+
+				Expect(c.Create(ctx, dns)).To(Succeed())
+				Expect(dnsRecord.Deploy(ctx)).To(Succeed())
+
+				deployedDNS := &extensionsv1alpha1.DNSRecord{}
+				err = c.Get(ctx, client.ObjectKey{Name: name, Namespace: namespace}, deployedDNS)
+				Expect(err).NotTo(HaveOccurred())
+
+				patch := client.MergeFrom(deployedDNS.DeepCopy())
+				deployedDNS.Spec.Values = []string{address, "8.8.8.8", "1.1.1.1"}
+				Expect(c.Patch(ctx, deployedDNS, patch)).To(Succeed())
+				Expect(dnsRecord.Deploy(ctx)).To(Succeed())
```

Comment:
I'm not sure, if I fully get what you are doing in this test case:
Why are you calling `Deploy` twice?
Also, you would need to get the `DNSRecord` again after deploying to assert the expected object.

Reply:
Trying to simulate the bug basically.

Reply:
Okay I think I had it backwards. Obviously we want to change the addresses of the `dnsRecord` so that the `gardenlet` can update the dnsRecord with the `reconcile` operation hopefully :P